thirdeye/thirdeye-frontend/app/pods/services/api/anomalies/service.js

```diff
@@ -0,0 +1,78 @@
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
+import EmberObject from '@ember/object';
+import { assert } from '@ember/debug';
+/**
+ * @type {Ember.Service}
+ * @summary This service provides all the api calls for anomalies related data. An `Ember.Service`
+   is a long-lived Ember object that can be made available in different parts of your application.
+ * @example anomaliesApiService: service('services/api/anomalies');
+ */
+export default Service.extend({
+  store: service('store'),
+  queryCache: service('services/query-cache'),
+
+  /**
+   * @summary Fetch all application names. We can use it to list in the select box.
+   * @method queryApplications
+   * @param {String} appName - the application name for creating the cacheKey
+   * @example: /thirdeye/entity/APPLICATION
+     usage: `this.get('anomaliesApiService').queryApplications();` //vs using `await this.get('store').findAll('application');`
+   */
+  async queryApplications(appName, start, end ) {
+    const queryCache = this.get('queryCache');
+    const modelName = 'application';
+    const query = { appName, start, end };
+    const cacheKey = queryCache.urlForQueryKey(modelName, {});//TODO: Won't pass all the `query` here. The `cacheKey` do not need to be uniqued, since all apps has the same list of apps.
+    const applications = await queryCache.query(modelName, query, { reload: false, cacheKey });
+    return applications;
+  },
+
+  /**
+   * @summary Fetch all anomalies by application name and start time
+   * @method queryAnomaliesByAppName
+   * @param {String} appName - the application name
+   * @param {Number} startStamp - the anomaly iso start time
+   * @return {Ember.RSVP.Promise}
+   * @example: for call `/userdashboard/anomalies?application={someAppName}&start={1508472800000}`
+     usage: `this.get('anomaliesApiService').queryAnomaliesByAppName(this.get('appName'), this.get('startDate'));`
+   */
+  async queryAnomaliesByAppName(appName, start) {
+    assert('you must pass appName param as an required argument.', appName);
+    assert('you must pass start param as an required argument.', start);
+
+    const queryCache = this.get('queryCache');
+    const modelName = 'anomalies';
+    const query = { application: appName, start };
+    const anomalies = await queryCache.query(modelName, query, { reload: false, cacheKey: queryCache.urlForQueryKey(modelName, query) });
+    return anomalies;
+    //const url = anomalyApiUrls.getAnomaliesByAppNameUrl(appName, startTime);//TODO: remove from the utils/api/anomaly.js - lohuynh
```

Comment:
I see. So this new architecture renders our API utils obsolete. One of the reasons for having those was to have a centralized inventory of all APIs used. What do you recommend for that?